### PR TITLE
Add Diagnostics.Middleware readme

### PR DESCRIPTION
**Pull request by Tratcher**
Still needs more usage details.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4534)

### CODE REVIEW

File: src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/README.md [hunk 1]
```diff
@@ -1 +1,88 @@
...
+
+var app = builder.Build();
+
+app.UseRequestCheckpoint();
+app.UseRequestLatencyTelemetry();
+```
+
+## HTTP Request Logs Enrichment and Redaction
+
+These components enable enriching and redacting ASP.NET Core's [HTTP request logs](https://learn.microsoft.com/aspnet/core/fundamentals/http-logging/).
```
Comment (joperezr) [inline src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/README.md:57]: I'm sure I'm missing something, but didn't @geeknoid said that we are no longer doing any enrichment in logging?

**Comment by joperezr** [comment]
Content is looking great 😃 

Not sure if you were planning to add it later, but usually package readmes have a section in the end for feedback pointing back to the repo.

File: src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/README.md (hunk 1 above)
Comment (Tratcher) [reply src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/README.md:57]: We're not doing general logging enrichment, but we're still doing http request logging enrichment.

File: src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/README.md [hunk 2]
```diff
@@ -1 +1,94 @@
...
+
+app.UseRequestCheckpoint();
+app.UseRequestLatencyTelemetry();
+```
+
+### HTTP Request Logs Enrichment and Redaction
+
+These components enable enriching and redacting ASP.NET Core's [HTTP request logs](https://learn.microsoft.com/aspnet/core/fundamentals/http-logging/).
+
+These APIs are only available for ASP.NET Core 8+.
```
Comment (joperezr) [inline src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/README.md:61]: This is true for the whole package, right? Although by taking a look at the project looks like this is also supported in .NET 6. Does this comment mean that tracking request latency is supported in .NET 6+ but doing Enrichment and Redaction is only for 8.0?

Comment (Tratcher) [reply src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/README.md:61]: > Does this comment mean that tracking request latency is supported in .NET 6+ but doing Enrichment and Redaction is only for 8.0?

Correct.

**Review by joperezr — APPROVED** [review]

**Pull request merged**